unit/pointer-analysis/custom_value_set_analysis.cpp

```diff
@@ -0,0 +1,387 @@
+/*******************************************************************\
+
+Module: Value-set analysis tests
+
+Author: Chris Smowton, chris@smowton.net
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <util/config.h>
+#include <langapi/mode.h>
+#include <goto-programs/initialize_goto_model.h>
+#include <goto-programs/goto_inline.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <java_bytecode/java_types.h>
+#include <pointer-analysis/value_set_analysis.h>
+
+/// An example customised value_sett. It makes a series of small changes
+/// to the underlying value_sett logic, which can then be verified by the
+/// test below:
+/// * Writes to variables with 'ignored' in their name are ignored.
+/// * All writes are made weak by forcing `add_to_sets` to true.
+/// * Never propagate values via the field "never_read"
+/// * Adds an ID_unknown to the value of variable "maybe_unknown", and
+///     to the possible referees of fields named unknown_field_ref and
+///     variables named unknown_global_ref
+class custom_value_sett:public value_sett
```

Comment:
You might consider pulling this class out into a separate file with just the above documentation left in the test file since it isn't directly relevant to the test. Definitely not a requirement, so do disregard if you disagree.